server/height_map.go

```diff
@@ -0,0 +1,78 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+// HeightMapper keeps track of which optimint blocks where posted to which celestia
+// blocks
+type HeightMapper struct {
+	Heights map[string]int64
+	mut     *sync.RWMutex
+}
```

Comment:
@tzdybal perhaps keeping track of optimint block height: celestia block height, belongs in optimint. I didn't want to add yet another KV database, but we will have to do something in the long term, as using a simple map and saving it as a json will obviously not be ideal for a blockchain with a few million blocks.

Reply:
I think keeping track is one of DALC functions, and Optimint should be agnostic. Also, I don't see any reason not to use KV store - it's operation won't be resource-consuming anyways. 

This implementation is going to work only for aggregator node (to be super specific: DALC instance that will be used by aggregator, will also be able to serve optimint block data to other nodes). I described in #24 results of our previous conversations about locating blocks in celestia chain.

IMHO we should merge as is and create follow-up issue.

Reply:
:+1: follow-up https://github.com/celestiaorg/dalc/issues/39